### PR TITLE
[ADMIN] fix path concat to support admin serving at / path

### PR DESCRIPTION
**Pull request by dekinderfiets**
### What does it do?

Fixed the concatenation of paths.
When `strapi.config.admin.path` value is `/` (According to [the docs](https://docs.strapi.io/dev-docs/admin-panel-customization#different-servers), it should be supported), `koa-static` serving throws an `AssertionError` since the path is cannot be registered because the registered route is`//:path*` instead of `/:path*`.

### Why is it needed?

It is needed so we can support serving the admin at `https://example.com/` and not only at `https://example.com/something`

### How to test it?

To reproduce the error, just set the `url` key at `config/admin(.ts/.js)` to `/` and start Strapi.

### Related issue(s)/PR(s)

A few days ago someone faced the same issue at the forums:
https://forum.strapi.io/t/routing-for-separate-subdomains-for-api-and-admin-page/29177

### CODE REVIEW

**Comment by alexiuscrow** [comment]
This PR looks really great. I think this would greatly improve path routing.
Thanks, @dekinderfiets!

File: packages/core/admin/server/routes/serve-admin-panel.js [hunk 1]
```diff
@@ -29,7 +29,7 @@ const registerAdminPanelRoute = ({ strapi }) => {
   strapi.server.routes([
     {
       method: 'GET',
-      path: `${strapi.config.admin.path}/:path*`,
+      path: join(strapi.config.admin.path, '/:path*'),
```
Comment (gu-stav) [inline packages/core/admin/server/routes/serve-admin-panel.js:32]: I think this won't work on windows machines, which are using `\` as separator and should use the [`URL`](https://developer.mozilla.org/en-US/docs/Web/API/URL/URL) API instead. 

```
new URL("/:path*", strapi.config.admin.path).pathname
```

Comment (derrickmehaffy) [reply packages/core/admin/server/routes/serve-admin-panel.js:32]: This is a URL path not a file path, which is universal on all OSes

Comment (gu-stav) [reply packages/core/admin/server/routes/serve-admin-panel.js:32]: Rhe URL API is what you'd want to build up URLs. This is not a file-path; therefore, it is just a coincidence `path.join` produces the same result.

I am not sure, but I think if you'd run node on a windows machine `path.join` probably uses backslash as a separator.

**Comment by derrickmehaffy** [comment]
I have a feeling this might break a few things in recent updates. @dekinderfiets have you tested this in a deployed environment with something like nginx and no-prefix base path? `/`

**Comment by derrickmehaffy** [comment]
> @derrickmehaffy
> 
> > I have a feeling this might break a few things in recent updates. @dekinderfiets have you tested this in a deployed environment with something like nginx and no-prefix base path? `/`
> 
> I tried it all the variations mentioned above:
> 
> strapi.config.admin.path value	result
> ""	"/:path*"
> "/"	"/:path*"
> "/foo"	"/foo/:path*"
> "/foo/"	"/foo/:path*"
> "foo/"	"/foo/:path*"
> "foo"	"/foo/:path*"
> with an Nginx instance (Kubernetes Ingress) and it worked. If you want a more stable and maintained solution for concatenating the url parts we can use [url-join](https://www.npmjs.com/package/url-join). What do you think?

Ah I don't think we need that package, I was more so worried about the various nginx proxy (or generic proxy application) handling.

**Comment by derrickmehaffy** [comment]
Need to get engineers approval, I'll mention it to them in our internal slack

**Comment by derrickmehaffy** [comment]
Experimental version for testing: `0.0.0-experimental.e85c31c67a29c380cea20e131f8c2b8c4e4fbe82`

**Comment by innerdvations** [comment]
This looks good for me and seemed to work fine, but since it's part of the core path routing if someone else can do some more thorough QA to double-check me to make sure we don't break anything as a side effect ( @derrickmehaffy do you have time?)  that would be great, then we can merge 🚀

**Comment by derrickmehaffy** [comment]
I got a response from the customer that included a more complete snippet of code that I want to try to adapt yet.

One question though, why don't we convert the core-store to a content type so we can avoid some of this weirdness here and be able to use the entityService on it?

I mean you can in theory use the query engine on it just fine as I did in this example: https://github.com/derrickmehaffy/strapi-example-v4-check-license-expire/blob/main/src/api/checkLicense/controllers/checkLicense.js

**Comment by derrickmehaffy** [comment]
Yeah I'll do some of my prod type testing with Nginx and caddy to see how well it works in a real world use-case.

**Comment by hanpaine** [comment]
Hey @dekinderfiets thanks for your patience! I've re-requested that review from Derrick; in the meantime, please may you point this at develop instead of strapi:main 🙏

**Comment by alexandrebodin** [comment]
> [!IMPORTANT] 
> With the release of Strapi 5, we have lifted the code freeze. To ensure we can effectively review and assist with contributions, we are now only accepting PRs targeting version 5 (the develop branch).<br/> 
> As your PR targets version 4, it has been automatically closed. However, we would greatly appreciate it if you could update your contribution to align with version 5 and resubmit!

**Pull request closed without merging**